modules/openid_auth/app/controllers/openid_auth/v0/mvi_users_controller.rb

```diff
@@ -45,22 +43,28 @@ def missing_loa
         request.headers['x-va-level-of-assurance'].blank?
       end
 
-      def build_identity
-        UserIdentity.create(uuid: request.headers['x-va-idp-uuid'],
-                            email: request.headers['x-va-user-email'],
-                            first_name: request.headers['x-va-first-name'],
-                            last_name: request.headers['x-va-last-name'],
-                            # TODO: break this out into a method that handles unknown gender
-                            gender: request.headers['x-va-gender']&.chars&.first&.upcase,
-                            birth_date: request.headers['x-va-dob'],
-                            ssn: request.headers['x-va-ssn'],
-                            mhv_icn: request.headers['x-va-mhv-icn'],
-                            dslogon_edipi: request.headers['x-va-dslogon-edipi'],
-                            loa:
-                            {
-                              current: request.headers['x-va-level-of-assurance'].to_i,
-                              highest: request.headers['x-va-level-of-assurance'].to_i
-                            })
+      def save_requested_identity
+        # In addition to constructing the the identity object, we save it in order to prime the
+        # identity cache. This is done because this endpoint is primarily called by the saml-proxy
+        # during the login process. If a user is logging in, usually a client app will make use of
```

Comment:
The OpenidUserIdentity derived from these headers is different than what is derived from the incoming openid token (or token+okta profile). I.e this OpenidUserIdentity will vary depending on the credential type - might just have an mhv_icn, might have an edipi, might have first/last/dob/ssn/gender. 
Whereas the OpenidUserIdentity from the token+profile should be entirely uniform and have the fixed set of traits from the okta profile. 
So I'm not sure priming the identity cache is actually that helpful? But as long as we understand how one overrides/merges-with the other, I don't see any obvious problem with this either.

Reply:
That's a good point that the underlying identity fields will/could have different values. I originally thought it didn't matter much what the stored identity attributes were since the token would be linked to the okta profile. However, it seems like we give API endpoints in `modules/` carte blanche access to the identity fields via `@current_user` so we should try to ensure they are as stable as possible. I'll remove this caching.

Reply:
It seems odd that there would be a separate controller for building this. Could this have been done as part of the initial callback in OpenIDUserIdentity?

Reply:
@saneshark the context for this controller is described here:
https://github.com/department-of-veterans-affairs/vets-contrib/pull/1377

This is what the SAML proxy calls to turn heterogenous traits into a single attribute (ICN) that we store in Okta for future token issuance. So it's really a first crack at extracting out the MVI logic into something usable outside of vets-api. It is not feasible for the input to be either a SAML assertion or an OpenID token at this point in the flow, because it's in the midst of converting one to the other, so we opted for a loose collection of headers (could have also been a request body :shrug:).

Reply:
I think there's a likely path forward where the user/session stuff gets extracted out further from vets-api, and becomes its own service, in which case we will likely refine this into a more generally applicable API.